### DISCUSSION TITLE
Zigbee only variant with cheaper MCU (E73-2G4M08S1C)

### DISCUSSION THREAD

**Discussion by BenRoe** [rbaron/b-parasite#174] [Ideas]
The Bluetooth/Zigbee module ([E73-2G4M08S1C](https://www.lcsc.com/mobile/product-detail/Bluetooth-Modules_Chengdu-Ebyte-Elec-Tech-E73-2G4M08S1C_C356849.html)) is the most expensive ($8,70) part.
I wonder if most people use the Bluetooth feature?
A Zigbee only variant can use the [E73-2G4M08S1C](https://www.lcsc.com/mobile/product-detail/ZigBee-Modules_Chengdu-Ebyte-Elec-Tech-E18-MS1-PCB_C97344.html) module ($3.21).

Sadly i do not have the knowledge and time to create a variant with this mcu.

**Comment by Hedda** [comment]
@BenRoe FYI, [b-parasite](https://github.com/rbaron/b-parasite) project code is written for the Nordic Semiconductor nRF52840 SoC/MCU using their nRF5 SDK and nRF Connect which in turn are based on the [Zephyr Project real-time operating system (RTOS)](https://github.com/zephyrproject-rtos/zephyr) so would likley be a lot of work porting to other SoC/MCUs that is not running a similar stack and is not supported by Zephyr. 

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/2.1.0-rc1/zephyr/hardware/porting/index.html

The E18-MS1-PCB module you linked to is based on the (old) Texas Instruments CC2530 which uses older TI's Z-Stack and SimpleLink SDK.

https://www.lcsc.com/product-detail/ZigBee-Modules_Chengdu-Ebyte-Elec-Tech-E18-MS1-PCB_C97344.html

If want to use CC253x recommend that instead check out existing projects that are written for Texas Instruments Z-Stack like example these:

https://github.com/diyruz/flower

https://github.com/smartboxchannel/Plant-Watering-Sensor-Zigbee

https://www.tindie.com/stores/diyberk/

Also check out PVTO Zigbee Switch Configurable Firmware which can make custom firmware for TI's CC2530, CC2531, CC2652, and CC1352:

https://ptvo.info/

https://ptvo.info/zigbee-configurable-firmware-features/

**Reply by BenRoe** [reply]
Many thanks for the detailed answer. This helps and makes totally sense.
Will check your mentioned links.
